src/intents/DSL/schema.ts

```diff
@@ -0,0 +1,111 @@
+import { TokenStandard } from "@brinkninja/types";
+import Joi from "joi";
+import { joi } from "../../internal/joiExtended";
+
+// Object.keys returns keys and values that's why we're filtering out the values using isNaN
+const tokenStandards = Object.keys(TokenStandard).filter(key => isNaN(Number(key)));
+
+export const TokenArgsSchema = joi.alternatives().try(
+  joi.ethereumAddress(),
+  joi.object({
+    address: joi.ethereumAddress().required(),
+    standard: joi.string().valid(...tokenStandards).optional(),
+    idsMerkleRoot: joi.string().optional(),
+    id: joi.bigIntish().optional(),
+    disallowFlagged: joi.boolean().optional(),
+  })
+);
+
+
+export const replaySchema = joi.object({
+  nonce: joi.bigIntish().min(1).required(),
+  runs: joi.string().valid('ONCE', 'UNTIL_CANCELLED').required(),
+});
+
+export const intervalConditionSchema = joi.object({
+  type: joi.string().valid('interval').required(),
+  id: joi.bigIntish().required(),
```

Comment:
these four params are not uint256. See https://github.com/brinktrade/brink-strategies/blob/31b2c01f510cbae01d13fca9364758cb25badb6b/src/Primitives/Primitives01.sol#L103 they are uint64, uint128, uint128, and uint16. Other DSL commands can also be checked against the primitives (will be renamed to segments) fns

Reply:
@phanpak still need to update this